Clang++ exited with code 1

<!-- 🚨 STOP 🚨 𝗦𝗧𝗢𝗣 🚨 𝑺𝑻𝑶𝑷 🚨 -->

<!--
Please fill in the *entire* template below.
The template is designed to avoid unnecessary delays to confirm and fix reported issues. Issues with missing information generally end up with a `need-info` label and take significantly more time to fix. Please help us help you!
For more information see https://github.com/xamarin/xamarin-macios/wiki/Submitting-Bugs-&-Suggestions
For support requests use https://www.xamarin.com/support
-->


### Steps to Reproduce

1. Create a new MAUI project on a **Windows Machine** with Visual Studio
2. Install the Branch-Xamarin-Linking-SDK NuGet package.
3. Pair the Windows device to an M1 Mac and try to run on a simulator or a device.

### Expected Behavior
App should build successfully 

### Actual Behavior

```
Undefined symbols for architecture arm64:
  "_OBJC_CLASS_$_BNCCommerceEvent", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCLinkCache", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCLinkData", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCPreferenceHelper", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCProduct", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCServerInterface", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCServerRequest", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCServerRequestQueue", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BNCServerResponse", referenced from:
     -u command line option
  "_OBJC_CLASS_$_Branch", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BranchActivityItemProvider", referenced from:
     -u command line option
  "_OBJC_CLASS_$_BranchCSSearchableItemAttributeSet", referenced from:
     -u command line opt		
```

### Environment

<!--
1. How to get version information: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#version-information
2. Paste into the code block below (between ```)
-->



<details>
<summary>Version information</summary>

```
Microsoft Visual Studio Enterprise 2022
Version 17.7.5
VisualStudio.17.Release/17.7.5+34202.233
Microsoft .NET Framework
Version 4.8.09037

Installed Version: Enterprise

ASP.NET and Web Tools   17.7.273.65229
ASP.NET and Web Tools

Azure App Service Tools v3.0.0   17.7.273.65229
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.7.273.65229
Azure Functions and Web Jobs Tools

C# Tools   4.7.0-3.23469.2+a3b9f0c274649870015cfe508cacde8fdc15df55
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.4.34 (main@d5ab18b)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Mono Debugging for Visual Studio   17.7.27 (547ea6f)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.7.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.7.3.2333001+0ab18affdf2a37647768d0e25f5f021bee6257a1
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.7.10.1
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.20829.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.7.0-3.23469.2+a3b9f0c274649870015cfe508cacde8fdc15df55
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.7.0-beta.23314.10+e612cf93b989503c89e3a5830090062b7ab5e143
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.7.0.223 (d17-7@c374b3e)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.7.3.9 (remotes/origin/d17-7@796d191def)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.7.21 (150ec9f)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   13.2.1.2 (d17-5/a8a26c7)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: d9a6e87
    Java.Interop: xamarin/java.interop/d17-5@149d70fe
    SQLite: xamarin/sqlite/3.40.1@68c69d8
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-5@ca1552d


Xamarin.iOS and Xamarin.Mac SDK   16.4.0.18 (9d266025e)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

</details>

### Build Logs
[msbuild.zip](https://github.com/xamarin/xamarin-macios/files/12928453/msbuild.zip)
<!--
1. How to get a build log: https://github.com/xamarin/xamarin-macios/wiki/Diagnosis#build-logs
2. Place the cursor below this comment block.
3. Drag and drop the build log.
-->

### Example Project (If Possible)

<!--


1. Place cursor below this comment block.
2. Drag and drop the compressed project or files needed to reproduce.
-->

<!--
Switch to the "Preview" tab to ensure your issue renders correctly.
-->

--- a/msbuild/Xamarin.MacDev.Tasks/Tasks/ResolveNativeReferencesBase.cs
+++ b/msbuild/Xamarin.MacDev.Tasks/Tasks/ResolveNativeReferencesBase.cs
@@ -526,7 +526,20 @@ namespace Xamarin.MacDev.Tasks {
 
 		public IEnumerable<ITaskItem> GetAdditionalItemsToBeCopied ()
 		{
-			return CreateItemsForAllFilesRecursively (NativeReferences);
+			var rv = new List<ITaskItem> ();
+			rv.AddRange (CreateItemsForAllFilesRecursively (NativeReferences));
+			foreach (var reference in References) {
+				var resourcesPackage = Path.Combine (Path.GetDirectoryName (reference.ItemSpec), Path.GetFileNameWithoutExtension (reference.ItemSpec)) + ".resources";
+				if (Directory.Exists (resourcesPackage)) {
+					var resources = CreateItemsForAllFilesRecursively (new string [] { resourcesPackage });
+					rv.AddRange (resources);
+					continue;
+				}
+				var zipPackage = resourcesPackage + ".zip";
+				if (File.Exists (zipPackage))
+					rv.Add (new TaskItem (zipPackage));
+			}
+			return rv;
 		}
 	}
 }

